erpnext/controllers/selling_controller.py

```diff
@@ -548,3 +548,50 @@ def update_project_billing_and_sales(self):
 			doc.set_gross_margin(update=True)
 			doc.set_status(update=True)
 			doc.notify_update()
+
+	@frappe.whitelist()
+	def set_rate_as_cost(self):
+		if not has_valuation_read_permission():
+			frappe.throw('User not allowed to read valuation rate')
```

Comment:
Always translate user facing strings with `_("...")`
Change the error to: `_("You do not have permission to get cost as rate")`